src/main/java/com/paneedah/mwc/groovyscript/script/CraftingStation.java

```diff
@@ -0,0 +1,177 @@
+package com.paneedah.mwc.groovyscript.script;
+
+import com.cleanroommc.groovyscript.api.GroovyLog;
+import com.cleanroommc.groovyscript.api.IIngredient;
+import com.cleanroommc.groovyscript.helper.recipe.AbstractRecipeBuilder;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import com.paneedah.mwc.groovyscript.MWCContainer;
+import com.paneedah.mwc.groovyscript.recipes.GSCrafting;
+import com.paneedah.weaponlib.crafting.CraftingEntry;
+import com.paneedah.weaponlib.crafting.CraftingGroup;
+import com.paneedah.weaponlib.crafting.CraftingRegistry;
+import com.paneedah.weaponlib.crafting.IModernCrafting;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.oredict.OreDictionary;
+
+import java.util.ArrayList;
+
+import static com.paneedah.weaponlib.crafting.CraftingGroup.*;
+import static com.paneedah.weaponlib.crafting.CraftingRegistry.*;
+
+public class CraftingStation extends VirtualizedRegistry<GSCrafting> {
+
+    @Override
+    public void onReload() {
+        this.removeScripted().forEach(CraftingRegistry::deleteRecipeRegistry);
+        this.restoreFromBackup().forEach(CraftingRegistry::registerRecipe);
+    }
+
+
+    //Nuke ALL recipes. no matter the category.
+
+    public void removeAll(){
+        clearRegistry();
+    }
+
+    //Remove Recipe that outputs given Ingredient.
+
+    public void remove(IIngredient ingredient){
+        for(ArrayList<IModernCrafting> list:craftingMap.values()){
+            for(IModernCrafting recipe: list){
+                if(ingredient.test(new ItemStack(recipe.getItem()))){
+                    deleteRecipeRegistry(recipe);
+                }
+            }
+        }
+    }
+
+    //Nuke ALL recipes of Given Category.
+
+    public void removeAllinGroup(CraftingGroup group){
+        clearGroup(group);
+    }
+
+    public void removeAllinGroup(String group){
+        clearGroup(CraftingGroup.valueOf(group));
+    }
+
+    public void removeinGroup(IIngredient ingredient, CraftingGroup group) {
+        for (IModernCrafting recipe : craftingMap.get(group)) {
+            if (ingredient.test(new ItemStack(recipe.getItem()))) {
+                deleteRecipeRegistry(recipe);
+            }
+        }
```

Comment:
One line if's an for's should not have curly brackets
```suggestion
        for (IModernCrafting recipe : craftingMap.get(group))
            if (ingredient.test(new ItemStack(recipe.getItem())))
                deleteRecipeRegistry(recipe);
```